### DISCUSSION TITLE
ElasticSearch ate all the disk space on the Manager

### DISCUSSION THREAD

**Discussion by 3isenHeiM** [Security-Onion-Solutions/securityonion#9058] [Unsupported Versions] [answered]
I'm running a distributed architecture, with 2 physical search nodes (2.5Tb disk each) and 1 virtual manager (500Gb disk). 

The manager got his disk full, and ElasticSearch couldn't start anymore. As a temporary patch, I've increase the disk size (+150Gb in the nsm LV), but l'd like to avoid this situation repeating itself in the future.

As it turned out, the `/nsm/elasticsearch/indices` folder takes all the available space.

We have several devices sending their data to filebeat of that manager, but I thought that the manager wasn't supposed to store that much data :

> that’s mainly used for storing [Cases](https://docs.securityonion.net/en/2.3/cases.html#cases) data and central configuration.`

We've reduces the thresholds of elasticsearch in `/opt/so/saltstack/local/pillar/global.sls`:
```
 hot_warm_enabled: False
 cluster_routing_allocation_disk.threshold_enabled: true
 cluster_routing_allocation_disk_watermark_low: '80%'
 cluster_routing_allocation_disk_watermark_high: '90%'
 cluster_routing_allocation_disk_watermark_flood_stage: '90%'
```

And in `/opt/so/saltstack/local/pillar/minions/<MANAGER>.sls`, we've set the limit at 200Gb:
```
log_size_limit: 200
```

**What else can I do to make sure that ES won't take all the available disk space again ?**

Also, I've read about Cross-Cluster Search in the doc, and it's the default configruation. What does this mean ?

> The manager node runs its own local copy of Elasticsearch, which manages cross-cluster search configuration for the deployment

Does the manger has a copy of the ES of the search nodes ?

**Comment by 3isenHeiM** [comment]
As it turns out, I have nearly 100 indices related to the playbooks: `so-playbook-alerts-YYY-MM-DD`, and most of them are around 2.5 GB.

Where can I limit the number of the playbooks kept ?

**Reply by 3isenHeiM** [reply]
This solved the issue #9739 to limit the alerts kept.

**Comment by defensivedepth** [comment, marked as answer]
What version are you running?

How many plays do you have active and are you seeing a large number of alerts from Plays every day?

**Reply by 3isenHeiM** [reply]
It's the latest version.
And yes we are seeing like 20 alerts, some with 1500+ events.


We've started to investigate, and **some playbook trigger but on the wrong data.**


For example a Cisco playbook triggers on Syslon event logs 🤷‍♂️

Do you know the syntax of the playbook in order to add a filter on the dataset before actually triggering ?

**Reply by defensivedepth** [reply]
How many plays do you have active? 

Here is how to add a custom filter to a Play: https://docs.securityonion.net/en/2.3/playbook.html#tuning-plays

However, if you are having that many false positives, it might be worth it to dig a bit deeper - what are the names of some of the Plays that are triggering large numbers of false positives?

**Reply by 3isenHeiM** [reply]
https://bradleyjkemp.dev/sigmadoc/rules/windows/raw_access_thread/sysmon_raw_disk_access_using_illegitimate_tools.yml/

This playbook is triggered more than 1M times

![image](https://user-images.githubusercontent.com/26417172/200861742-b82d9961-364d-4ea0-ab14-6c335af4fdc6.png)


And the drilldown of this alert reveals that the matching log is ... zeek (DNS logs) 🤯🤯🤯

![image](https://user-images.githubusercontent.com/26417172/200862400-a8727f64-b386-4f57-aa3f-485c2eec8b1d.png)

**Reply by defensivedepth** [reply]
@3isenHeiM Yes, that has known FP issues. see here: https://github.com/Security-Onion-Solutions/securityonion/discussions/8105#discussioncomment-2946300

I would suggest disabling it.

**Reply by 3isenHeiM** [reply]
Thanks fro the prompt reply, will do.
